src/util/Utils.cpp

```diff
@@ -666,3 +666,18 @@ int Utils::createDatabaseFromDDL(const char *dbLocation, const char *ddlFileLoca
     util_logger.info("Database created successfully");
     return 0;
 }
+
+bool Utils::fileExistsWithReadPermission(const string &path) {
+    std::ifstream file(path);
+    return file.is_open();
```

Comment:
```suggestion
    return access(fileName.c_str(), R_OK) == 0;
```

We do not need to open the file to check for permissions

Reply:
Done.

Reply:
Ok